src/plugin/vault/vault_test.go

```diff
@@ -225,51 +226,188 @@ func TestLogin(t *testing.T) {
 	}
 }
 
+// TestLoginExpires tests two features:
+// 1. Returned Pachyderm tokens are revoked when their vault lease expires
+// 2. If a TTL is set in the plugin config and not in the login request, then
+//    the TTL from the config is applied to the login token
 func TestLoginExpires(t *testing.T) {
 	c, _, secret := loginHelper(t, "2s")
 
+	// Make sure token is valid
 	_, err := c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
 
-	time.Sleep(time.Duration(secret.Auth.LeaseDuration+1) * time.Second)
+	// Wait for TTL to expire and check that token is no longer valid
+	time.Sleep(time.Duration(secret.LeaseDuration+1) * time.Second)
+	_, err = c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
+	if err == nil {
+		t.Fatalf("API call should fail, but token did not expire")
+	}
+}
+
+// TestLoginTTLParam tests that if TTL is set in the request, make sure the
+// returned token has that TTL
+func TestLoginTTLParam(t *testing.T) {
+	vaultClientConfig := vault.DefaultConfig()
+	vaultClientConfig.Address = vaultAddress
+	v, err := vault.NewClient(vaultClientConfig)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	v.SetToken("root")
+
+	err = configurePlugin(t, v, "")
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	vl := v.Logical()
+	secret, err := vl.Write(
+		fmt.Sprintf("/%v/login/github:bogusgithubusername", pluginName),
+		map[string]interface{}{"ttl": "2s"},
+	)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if secret.LeaseDuration != 2 {
```

Comment:
So per offline discussion - response `LeaseDuration` is the remaining time left on the lease. That's why we have bounds in other tests below. So here to keep this non flaky maybe we should check its <=2 and >0

Reply:
Fixed